### PR TITLE
Replace Vagrant with Docker for idp-fixture

### PR DESCRIPTION
**Pull request by jkakavas**
The change replaces the Vagrant box based fixture with a fixture
based on docker compose and 2 docker images, one for an openldap
server and one for a Shibboleth SAML Identity Provider.

The configuration of both openldap and shibboleth is identical to
the previous one, in order to minimize required changes in the
tests

### CODE REVIEW

File: x-pack/test/idp-fixture/build.gradle [hunk 1]
```diff
@@ -1,41 +1,5 @@
...
-task destroy(type: org.elasticsearch.gradle.vagrant.VagrantCommandTask) {
-    command 'destroy'
-    args '-f'
-    boxName box
-    environmentVars vagrantEnvVars
-    dependsOn halt
-}
+apply plugin: 'elasticsearch.test.fixtures'
 
 thirdPartyAudit.enabled = false
```
Comment (atorok) [inline x-pack/test/idp-fixture/build.gradle]: The test-fixtures plugin takes care of these, so it shouldn't be required to explicitly disable tasks.

File: x-pack/test/idp-fixture/idp/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM unicon/shibboleth-idp:3.4.2
+
+COPY shibboleth-idp/ /opt/shibboleth-idp/
```
Comment (atorok) [inline x-pack/test/idp-fixture/idp/Dockerfile]: Why not pass these in as volumes? 
We could avoid building an image here making it slightly faster.

**Review by alpar-t — COMMENTED** [review]
Thanks for taking this on! 
I left a few comments.

File: x-pack/test/idp-fixture/build.gradle (hunk 1 above)
Comment (jkakavas) [reply x-pack/test/idp-fixture/build.gradle]: Awesome, didn't know that, will adjust

File: x-pack/test/idp-fixture/idp/Dockerfile (hunk 2 above)
Comment (jkakavas) [reply x-pack/test/idp-fixture/idp/Dockerfile]: `shibboleth-idp` would work as I'm copying all necessary files, but with the `shib-jetty-base` I'm basically overwriting one file ( `shib-jetty-base/start.d/ssl/ini` ) in the container and I want to use the existing `shib-jetty-base` from the image for the rest. 

I could possibly get the `shib-jetty-base/` out of the image and rewrite the ssl.ini file and pass it as a volume if you think this is preferable.

Comment (atorok) [reply x-pack/test/idp-fixture/idp/Dockerfile]: you _can_  mount a single file as a volume, so if it's only the `ini` you care about just append `/ini` to both ends of the volume def

**Review by alpar-t — APPROVED** [review]
LGTM

**Comment by atorok** [comment]
Relates to  #34095

**Pull request merged**